torchtext/experimental/vocab.py

```diff
@@ -71,6 +71,23 @@ def vocab_from_file(file_object, min_freq=1, unk_token='<unk>', num_cpus=4):
     return Vocab(vocab_obj)
 
 
+def build_vocab_from_iterator(iterator):
+    """
+    Build a Vocab from an iterator.
+
+    Arguments:
+        iterator: Iterator used to build Vocab. Must yield list or iterator of tokens.
+    """
+
+    counter = Counter()
+    for tokens in iterator:
+        counter.update(tokens)
+    sorted_by_freq_tuples = sorted(counter.items(), key=lambda x: x[1], reverse=True)
```

Comment:
Is sorting of the sequences a requirement / necessity? If yes, why not move it to `vocab` itself?

Reply:
We could. It's just the design of the vocab API.

Reply:
We should always sort the tokens by frequency so that tokens that are accessed frequently are close to each other in the hash table, which can improve caching.

Reply:
We do sorting the tokens by frequency here when constructs a vocab instance. Francisco's suggestion is to move the sorting step to the vocab itself, not staying outside.